packages/git/src/electron-node/askpass/askpass.ts

```diff
@@ -0,0 +1,205 @@
+        const chunks: string[] = [];
+        req.setEncoding('utf8');
+        req.on('data', (d: string) => chunks.push(d));
+        req.on('end', () => {
+            const { request, host } = JSON.parse(chunks.join(''));
+
+            this.prompt(host, request).then(result => {
+                res.writeHead(200);
+                res.end(JSON.stringify(result));
+            }, () => {
+                res.writeHead(500);
+                res.end();
+            });
+        });
+    }
+
+    protected async prompt(host: string, request: string): Promise<string> {
+        try {
+            const answer = await this.promptServer.ask({
+                password: /password/i.test(request),
+                text: request,
+                details: `Git: ${host} (Press 'Enter' to confirm or 'Escape' to cancel.)`
+            });
+            if (GitPrompt.Success.is(answer) && typeof answer.result === 'string') {
+                return answer.result;
+            } else if (GitPrompt.Cancel.is(answer)) {
+                return '';
+            } else if (GitPrompt.Failure.is(answer)) {
+                const { error } = answer;
+                throw error;
+            }
+            throw new Error('Unexpected answer.'); // Do not ever log the `answer`, it might contain the password.
+        } catch (e) {
+            this.logger.error(`An unexpected error occurred when requesting ${request} by ${host}.`, e);
+            return '';
+        }
+    }
+
+    protected async randomBytes(size: number): Promise<Buffer> {
+        return new Promise<Buffer>((resolve, reject) => {
+            crypto.randomBytes(size, (error: Error, buffer: Buffer) => {
+                if (error) {
+                    reject(error);
+                    return;
+                }
+                resolve(buffer);
+            });
+        });
+    }
+
+    async getEnv(): Promise<AskpassEnvironment> {
+        if (!this.enabled) {
+            return {
+                GIT_ASKPASS: path.join(__dirname, '..', '..', '..', 'src', 'electron-node', 'askpass', 'askpass-empty.sh')
+            };
+        }
+
+        const [
+            ELECTRON_RUN_AS_NODE,
+            GIT_ASKPASS,
+            THEIA_GIT_ASKPASS_NODE,
+            THEIA_GIT_ASKPASS_MAIN,
+            THEIA_GIT_ASKPASS_HANDLE
+        ] = await Promise.all([
+            this.ELECTRON_RUN_AS_NODE,
+            this.GIT_ASKPASS,
+            this.THEIA_GIT_ASKPASS_NODE,
+            this.THEIA_GIT_ASKPASS_MAIN,
+            this.THEIA_GIT_ASKPASS_HANDLE
+        ]);
+
+        return {
+            ELECTRON_RUN_AS_NODE,
+            GIT_ASKPASS,
+            THEIA_GIT_ASKPASS_NODE,
+            THEIA_GIT_ASKPASS_MAIN,
+            THEIA_GIT_ASKPASS_HANDLE
+        };
+    }
+
+    dispose(): void {
+        this.server.close();
+        if (this.ipcHandlePath && !isWindows) {
+            fs.unlinkSync(this.ipcHandlePath);
```

Comment:
@kittaakos, I'm getting the following logged on exit.

`root ERROR Error: ENOENT: no such file or directory, unlink '/var/folders/wl/7h9ks5m96jqc0c861k9jfrvc0000gn/T/theia-git-askpass-8553a957cf7af4af078a12bb48be63daab69cbdf-sock'`

is it uncritical? I mean, because it's when I close the App.

Reply:
How did you "exit"? Presumably, it happened in the electron application, right?

Reply:
Right. Just closed the App. (Did not send SIG_INT)

Reply:
I have fixed it.